SynonymGroup.ts

```diff
@@ -1,192 +1,736 @@
+          a dwcFP:TaxonConcept .
+      OPTIONAL { ?aug trt:augmentsTaxonConcept ?tc . }
+      OPTIONAL { ?def trt:definesTaxonConcept ?tc . }
+      OPTIONAL { ?dpr trt:deprecates ?tc . }
+      OPTIONAL { ?cite cito:cites ?tc . }
+    }
+  }
+}
+GROUP BY ?tn ?tc ?col ?rank ?genus ?species ?infrasp ?name ?authority
+LIMIT 500`;
+    // For unclear reasons, the query breaks if the limit is removed.
+
+    if (this.controller.signal?.aborted) return Promise.reject();
+
+    /// ?tn ?tc !rank !genus ?species ?infrasp !name !authority ?tcAuth
+    const json = await this.sparqlEndpoint.getSparqlResultSet(
+      query,
+      { signal: this.controller.signal },
+      `NameFromCol ${colUri}`,
+    );
+
+    return this.handleName(json, justification);
+  }
+
+  /** @internal */
+  private async getNameFromTC(
+    tcUri: string,
+    justification: Justification,
+  ): Promise<void> {
+    // Note: this query assumes that there is no sub-species taxa with missing dwc:species
+    // Note: the handling assumes that at most one taxon-name matches this colTaxon
+    const query = `
+PREFIX dwc: <http://rs.tdwg.org/dwc/terms/>
+PREFIX dwcFP: <http://filteredpush.org/ontologies/oa/dwcFP#>
+PREFIX cito: <http://purl.org/spar/cito/>
+PREFIX trt: <http://plazi.org/vocab/treatment#>
+SELECT DISTINCT ?tn ?tc ?col ?rank ?genus ?species ?infrasp ?name ?authority
+  (group_concat(DISTINCT ?tcauth;separator=" / ") AS ?tcAuth)
+  (group_concat(DISTINCT ?aug;separator="|") as ?augs)
+  (group_concat(DISTINCT ?def;separator="|") as ?defs)
+  (group_concat(DISTINCT ?dpr;separator="|") as ?dprs)
+  (group_concat(DISTINCT ?cite;separator="|") as ?cites)
+  (group_concat(DISTINCT ?trtn;separator="|") as ?tntreats)
+  (group_concat(DISTINCT ?citetn;separator="|") as ?tncites) WHERE {
+  <${tcUri}> trt:hasTaxonName ?tn .
+  ?tc trt:hasTaxonName ?tn ;
+      dwc:scientificNameAuthorship ?tcauth ;
+      a dwcFP:TaxonConcept .
+
+  ?tn a dwcFP:TaxonName .
+  ?tn dwc:rank ?rank .
+  ?tn dwc:kingdom ?kingdom .
+  ?tn dwc:genus ?genus .
+  OPTIONAL {
+    ?tn dwc:species ?species .
+    OPTIONAL { ?tn dwc:subspecies|dwc:variety ?infrasp . }
+  }
+  
+  OPTIONAL {
+    ?col dwc:taxonRank ?rank .
+    OPTIONAL { ?col dwc:scientificNameAuthorship ?colAuth . }
+    ?col dwc:scientificName ?fullName . # Note: contains authority
+    ?col dwc:genericName ?genus .
+    # TODO # ?col dwc:parent* ?p . ?p dwc:rank "kingdom" ; dwc:taxonName ?kingdom .
+
+    {
+      ?col dwc:specificEpithet ?species .
+      ?tn dwc:species ?species .
+      {
+        ?col dwc:infraspecificEpithet ?infrasp .
+        ?tn dwc:subspecies|dwc:variety ?infrasp .
+      } UNION {
+        FILTER NOT EXISTS { ?col dwc:infraspecificEpithet ?infrasp . }
+        FILTER NOT EXISTS { ?tn dwc:subspecies|dwc:variety ?infrasp . }
+      }
+    } UNION {
+      FILTER NOT EXISTS { ?col dwc:specificEpithet ?species . }
+      FILTER NOT EXISTS { ?tn dwc:species ?species . }
+    }
+  }
```

Comment:
In general, a problem is that this query wuth the UNIONs is much slower on lindas than on our server; however making them optional and FILTERing with
```

    OPTIONAL {
      ?col dwc:specificEpithet ?species .
      ?tn dwc:species ?tnspecies .
      OPTIONAL {
        ?col dwc:infraspecificEpithet ?infrasp .
        ?tn dwc:subspecies|dwc:variety ?tninfrasp .
      }
    }

    FILTER (COALESCE(?tnspecies, "N A") = COALESCE(?species, "N A"))
    FILTER (COALESCE(?tninfrasp, "N A") = COALESCE(?infrasp, "N A"))
```

does not work at all on our server, but is quite fast on lindas.

I am considering having both query-variants, and selecting depending on the endpoint which one to use. This should be SynoLib-user overridabe, but have hardcoded defaults for known servers (i.e. for those on the synospecies settings)